src/StateManager.ts

```diff
@@ -1792,6 +1863,90 @@ export default class StateManager {
     StateManager._nextStateId = 0;
     UndoRedoManager.reset();
   }
+  static async waitForLeftOrRightKey(): Promise<number> {
+    return new Promise((resolve) => {
+      function keyHandler(event: KeyboardEvent) {
+        console.log("Key detected:", event.code);
+        if (event.code === "ArrowLeft") {
+          resolve(-1);
+          cleanup();
+        } else if (event.code === "ArrowRight") {
+          resolve(1);
+          cleanup();
+        } else if (event.code === "Enter") {
+          resolve(-10);
+          cleanup();
+        } else {
+          resolve(0);
+          cleanup();
+        }
+      }
+
+      function buttonHandler() {
+        console.log("I was clicked");
+        resolve(-10);
+        cleanup();
+      }
+
+      function debugChangeHandler() {
+        console.log("Debug mode switched, resolving with -10");
+        resolve(-10);
+        cleanup();
+      }
+
+      function cleanup() {
+        window.removeEventListener("keydown", keyHandler);
+        StateManager.unsubscribe(debugChangeHandler);
+        if (testMachineButton) {
+          testMachineButton.removeEventListener("click", buttonHandler);
+        }
+      }
+
+      const testMachineButton = document.getElementById("testMachineButton");
+
+      if (testMachineButton) {
+        testMachineButton.addEventListener("click", buttonHandler);
+      }
+
+      window.addEventListener("keydown", keyHandler);
+      StateManager.subscribe(debugChangeHandler); // Listen for debug mode changes
+    });
+  }
+
+  public static delay(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+  public static async debugMachine(path: any[]) {
+    let flagtest = 1;
+    StateManager.deselectAllObjects();
+    let pos = 0;
+
+    while (1 == 1) {
+      if (pos >= path.length) {
+        pos = path.length - 1;
+        flagtest = 0;
+      }
+      if (pos < 0) {
+        pos = 0;
+        flagtest = 0;
+      }
+      for (const n of StateManager._nodeWrappers) {
+        if (n.labelText === path[pos] && flagtest == 1) {
+          StateManager.deselectAllObjects();
+          await StateManager.delay(100);
+          StateManager.selectObject(n);
+        }
+      }
+      let moveVal = await StateManager.waitForLeftOrRightKey();
+      flagtest = 1;
+      if (moveVal == -10) {
```

Comment:
We should try to avoid magic numbers like `-10` here. Perhaps instead an enum value could be passed out of `waitForLeftOrRightKey()`.